packages/bot/src/transformers/voiceState.ts

```diff
@@ -3,16 +3,17 @@ import type { Bot } from '../index.js'
 import { VoiceStateToggles } from './toggles/voice.js'
 
 export function transformVoiceState(bot: Bot, payload: { voiceState: DiscordVoiceState } & { guildId: bigint }): VoiceState {
-  const voiceState = {
-    toggles: new VoiceStateToggles(payload.voiceState),
+  const props = bot.transformers.desiredProperties.voiceState
+  const voiceState = {} as VoiceState
 
-    requestToSpeakTimestamp: payload.voiceState.request_to_speak_timestamp ? Date.parse(payload.voiceState.request_to_speak_timestamp) : undefined,
-    sessionId: payload.voiceState.session_id,
-
-    channelId: payload.voiceState.channel_id ? bot.transformers.snowflake(payload.voiceState.channel_id) : undefined,
-    guildId: payload.guildId || (payload.voiceState.guild_id ? bot.transformers.snowflake(payload.voiceState.guild_id) : 0n),
-    userId: payload.voiceState.user_id ? bot.transformers.snowflake(payload.voiceState.user_id) : 0n,
-  } as VoiceState
+  if (props.requestToSpeakTimestamp && payload.voiceState.request_to_speak_timestamp)
+    voiceState.requestToSpeakTimestamp = Date.parse(payload.voiceState.request_to_speak_timestamp)
+  if (props.channelId && payload.voiceState.channel_id) voiceState.channelId = bot.transformers.snowflake(payload.voiceState.channel_id)
+  if (props.guildId)
+    voiceState.guildId = payload.guildId || (payload.voiceState.guild_id ? bot.transformers.snowflake(payload.voiceState.guild_id) : 0n)
```

Comment:
```suggestion
    voiceState.guildId = payload.guildId
```

First of all, using `||` is not a great idea because of javascript and it's faulsy values (even tho it does not apply in this situation (or at least shouldn't)), second `payload.guild` is always present based on the typings for it

Reply:
I kept `||` to keep the functionality the same after the PR, even if it could have been wrong. I think converting these "otherwise if falsey" checks would deserve its own PR if we're going in that direction.

Reply:
Uh, i did not see at all that it was already in that way, so yeah that could have it's own pr, sorry

Reply:
All good, thank you